python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -137,15 +136,21 @@ private PythonType convertToFunctionType(FunctionSymbol symbol, Map<Symbol, Pyth
     return functionType;
   }
 
-  private static ParameterV2 convertParameter(FunctionSymbol.Parameter parameter) {
-    return new ParameterV2(parameter.name(),
-      PythonType.UNKNOWN,
-      parameter.hasDefaultValue(),
-      parameter.isKeywordOnly(),
-      parameter.isPositionalOnly(),
-      parameter.isKeywordVariadic(),
-      parameter.isPositionalVariadic(),
-      null);
+  private PythonType resolveReturnType(Map<Symbol, PythonType> createdTypesBySymbol, @Nullable ClassSymbol classSymbol) {
+    var returnType = PythonType.UNKNOWN;
```

Comment:
do you need to define the variable here? seems that in the condition below you can simply return `UNKNOWN` and define the `returnType` on the line 149 or simply return the optional chain